awswrangler/lakeformation/_read.py

```diff
@@ -0,0 +1,192 @@
+"""Amazon Lake Formation Module gathering all read functions."""
+import concurrent.futures
+import itertools
+import logging
+from typing import Any, Dict, Iterator, List, Optional, Tuple, Union
+
+import boto3
+import pandas as pd
+from pyarrow import NativeFile, RecordBatchStreamReader
+
+from awswrangler import _utils, exceptions
+from awswrangler._config import apply_configs
+from awswrangler.lakeformation._utils import wait_query
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+def _execute_query(
+    query_id: str,
+    token_work_unit: Tuple[str, int],
+    boto3_session: Optional[boto3.Session] = None,
+) -> pd.DataFrame:
+    client_lakeformation: boto3.client = _utils.client(service_name="lakeformation", session=boto3_session)
+    token, work_unit = token_work_unit
+    messages: NativeFile = client_lakeformation.execute(QueryId=query_id, Token=token, WorkUnitId=work_unit)["Messages"]
+    return RecordBatchStreamReader(messages.read()).read_pandas()
+
+
+def _resolve_sql_query(
+    query_id: str,
+    chunked: Optional[bool] = None,
+    use_threads: bool = True,
+    boto3_session: Optional[boto3.Session] = None,
+) -> Union[pd.DataFrame, Iterator[pd.DataFrame]]:
+    client_lakeformation: boto3.client = _utils.client(service_name="lakeformation", session=boto3_session)
+
+    wait_query(query_id=query_id, boto3_session=boto3_session)
+
+    # The LF Query Engine distributes the load across workers
+    # Retrieve the tokens and their associated work units until NextToken is ''
+    # One Token can span multiple work units
+    # PageSize determines the size of the "Units" array in each call
+    # TODO: Inquire about good page size # pylint: disable=W0511
+    scan_kwargs: Dict[str, Union[str, int]] = {"QueryId": query_id, "PageSize": 2}
+    next_token: str = "init_token"  # Dummy token
```

Comment:
Should it be `Optional[str]`?

Reply:
Hmm this string should either take a value or is equal to '' when you reach the last next_token (but never None). Would it still be `Optional[str]` in that case?

Reply:
oh I thought it would return `None`. So it is correct the way it is.